monai/data/dataset.py

```diff
@@ -237,7 +239,12 @@ class CacheDataset(Dataset):
     """
 
     def __init__(
-        self, data, transform: Callable, cache_num: int = sys.maxsize, cache_rate: float = 1.0, num_workers: int = 0
+        self,
```

Comment:
I think the cache dataset is to accelerate the transforms, supporting None transform is quite confusing. why do we need this?

Reply:
Because some other datasets are based on this CacheDataset, if no transforms, it should also work fine, right?
Just like `cache_num = 0`.
What do you think?

Reply:
And all the other datasets in MONAI can support None transform, I think we should keep the same logic for them?

Reply:
IMO the high-level dataset as an extension of these basic datasets should handle these cases. by default the basic cache datasets should require a user-specified transform. and it seems the current CacheDataset works fine if the user input to transform is None

Reply:
> And all the other datasets in MONAI can support None transform, I think we should keep the same logic for them?
support None and default to None are different here...

Reply:
OK, I see.
I didn't realize that the `Compose` generates an empty list for None:
```py
if transforms is None:
    transforms = []
```
I will revert the modification to `CacheDataset` and only add the unit tests for None transform.
And I will also remove the `None` default value for `PersistentDataset`.

Thanks.

Reply:
thank you, I think the unit tests for None transforms looks nice

Reply:
updated in the latest commit.
Thanks.